designs/2019-additional-lint-targets/README.md

```diff
@@ -0,0 +1,123 @@
+- Start Date: 2019-05-12
+- RFC PR: https://github.com/eslint/rfcs/pull/20
+- Authors: Toru Nagashima ([@mysticatea](https://github.com/mysticatea))
+
+# Configuring Additional Lint Targets with `.eslintrc`
+
+## Summary
+
+This proposal adds the ability to specify additional target files into configuration files. This enhancement will solve the pain that people have to use the `--ext` option with wanted file extensions even if they use plugins which support additional file types.
+
+## Motivation
+
+People have to use the `--ext` option or glob patterns to check wanted files even if they use plugins which support additional file types.
+
+```yml
+plugins:
+  - markdown
+  - html
+  - "@typescript-eslint"
+  - react
+  - vue
+```
+
+```bash
+# ESLint checks only `*.js`.
+eslint src docs
+
+# Needs `--ext` option
+eslint src docs --ext .js,.md,.html,.ts,.jsx,.tsx,.vue
+```
+
+However, using plugins which support additional file types is the intention that wants to check those files. The requirement of redundant CLI options is not reasonable.
+
+Per "[Related Discussions](#related-discussions)" section, this is very regularly requested; I want to configure additional target files with `.eslintrc`.
+
+## Detailed Design
+
+This proposal enhances `overrides` property of the config file.
+
+- If a config file in a directory has `overrides` property, ESLint checks the files which are matched by any of override entries (i.e., `files`/`excludedFiles` criteria) additionally in the directory.
+    - If any of `files` value of an override entry ended with `*`, this enhancement doesn't use the entry in order to avoid checking too many kinds of files.
+- This enhancement affects only the case where a directory path is provided on the CLI. If people provide glob patterns on the CLI, ESLint behaves the same as currently.
```

Comment:
Just to clarify, does this mean that this feature won't work if someone has an npm script like `eslint **/*.ts` with the following configuration?

```json
{
  "overrides": [
    {
      "files": ["*.ts"],
      "parser": "@typescript-eslint/parser"
    }
  ]
}
```

Reply:
In this case, ESLint will use the `@typescript-eslint/parser` package as the parser.

Reply:
Yes. If globs were used on CLI, ESLint works as same as currently. The target files are the files that the globs matched regardless of `overrides` settings.

On the other hand, if a directory path was used on CLI, ESLint will use `overrides[].files` instead of `--ext` option to search target files in the directory.

Reply:
I wonder if it will be confusing to the end user if running `eslint "lib/**/*"` and `eslint lib` produce different results by being run on different files.

Reply:
The current ESLint does different already.

- `eslint "lib/**/*"` checks all kinds of files in `lib`.
- `eslint lib` checks `*.js` files in `lib`.

Reply:
Doesn’t running`eslint "lib/**/*"` require the `ext` CLI option to actually check other file extensions though?

Maybe I’m misunderstanding something, but with this new feature, it seems to me like the above two examples could (and I think should) effectively be the same thing (when this new feature is being used in a config).

Reply:
ESLint doesn't use `--ext` option at all if globs ware given on CLI. ESLint checks all files the globs matched.

Reply:
Ah, that’s what I was missing. Thank you for explaining!